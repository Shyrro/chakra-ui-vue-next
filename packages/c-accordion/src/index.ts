import { h, defineComponent, PropType } from "vue"
import { chakra, DOMElements } from "@chakra-ui/vue-system"

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: "div",
    },
  },
  setup(props, { slots, attrs }) {
    return () =>
      h(chakra(props.as, { label: "accordion" }), { ...attrs }, slots)
  },
})

export default CAccordion
