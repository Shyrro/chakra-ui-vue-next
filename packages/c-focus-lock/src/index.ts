export * from './c-focus-lock'
