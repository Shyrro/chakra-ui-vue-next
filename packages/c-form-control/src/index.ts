export * from './c-form-control'
export * from './use-form-control'
export * from './c-form-error'