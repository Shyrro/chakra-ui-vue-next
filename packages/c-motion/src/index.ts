export * from './c-motion'
export * from './animate-presence'
export * from './motion-utils'
