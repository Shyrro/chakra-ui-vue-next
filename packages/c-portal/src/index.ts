export { default as CPortal } from "./portal"
