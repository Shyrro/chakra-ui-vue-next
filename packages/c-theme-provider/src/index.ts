import {
  h,
  defineComponent,
  Fragment,
  PropType,
  provide,
  inject,
  computed,
} from "vue"
import { Theme } from "@chakra-ui/vue-theme"

const CThemeProvider = defineComponent({
  name: "CThemeProvider",
  props: {
    value: {
      type: [Object] as PropType<Theme>,
      default: () => undefined,
    },
  },
  setup(props, { slots }) {
    const pluginTheme = inject("$chakraTheme")
    const applicationTheme = computed(() => props.value || pluginTheme)
    provide("$chakraTheme", applicationTheme.value)
    return () => h(Fragment, slots.default?.({ $chakraTheme: props.value }))
  },
})

export default CThemeProvider
