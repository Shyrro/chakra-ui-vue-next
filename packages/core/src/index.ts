import { computed, Plugin, Ref, ref, watch } from 'vue'
import defaultTheme, { ColorMode } from '@chakra-ui/vue-theme'
import { toCSSVar, WithCSSVar } from '@chakra-ui/styled-system'
import { chakra, injectGlobal, css } from '@chakra-ui/vue-system'
import internalIcons from './icon.internals'
import { extendTheme, ThemeOverride } from './extend-theme'
import { MergedIcons, parseIcons } from './parse-icons'
import { mode } from '@chakra-ui/vue-theme-tools'

interface ExtendIconsPath {
  path: string
  viewBox?: string
}

interface IconsOptions {
  pack?: 'fa' | 'fe'
  library?: {}
  extend?: Record<string, ExtendIconsPath>
}
export interface ChakraUIVuePluginOptions {
  cssReset?: boolean
  extendTheme?: ThemeOverride
  icons?: IconsOptions
  defaultColorMode?: ColorMode
}

const ChakraUIVuePlugin: Plugin = {
  install(app, options: ChakraUIVuePluginOptions = { cssReset: true }) {
    // 1. Get theme value
    // 2. Parse theme tokens to CSS variables
    // 3. Inject all CSS variables as theme object
    const theme = options.extendTheme || defaultTheme
    const computedTheme = computed<WithCSSVar<ThemeOverride>>(() =>
      toCSSVar(theme)
    )
    injectGlobal({
      ':root': computedTheme.value.__cssVars,
    })

    let libraryIcons = options.icons?.library || {}
    let extendedIcons = options.icons?.extend || {}

    // Initialize colormode
    const colorMode: ColorMode = theme.config?.initialColorMode || 'light'

    // Bind theme to application global properties and provide to application
    app.config.globalProperties.$chakraTheme = computedTheme.value
    app.provide('$chakraTheme', computedTheme.value as ThemeOverride)

    // Provide initial colormode
    app.config.globalProperties.$initialColorMode = colorMode
    const colorModeRef = ref<ColorMode>(colorMode)
    app.provide<Ref<ColorMode>>('$chakraColorMode', colorModeRef)

    libraryIcons = parseIcons(libraryIcons)

    // Merge internal icons and library icons
    const mergedIcons: MergedIcons = {
      ...internalIcons,
      ...libraryIcons,
      ...extendedIcons,
    }

    app.provide('$chakraIcons', mergedIcons)

    // Set color mode property
    app.config.globalProperties.$mode = mode
  },
}

export interface ThemeProviderProps extends ThemeOverride {}
export default ChakraUIVuePlugin
export { extendTheme }

// Export chakra factory function
export { chakra }

/**
 *
 * Component exports
 * ==================
 *
 * Dear contributors,
 *
 * Please keep these exports in Alphabetical order :)
 */

// A
export * from '@chakra-ui/c-accordion'
export * from '@chakra-ui/c-alert'

// B
export * from '@chakra-ui/c-button'

// C
export * from '@chakra-ui/c-color-mode'
export * from '@chakra-ui/c-close-button'
export * from '@chakra-ui/c-code'

// F
export * from '@chakra-ui/c-flex'
export * from '@chakra-ui/c-focus-lock'

// I
export * from '@chakra-ui/c-icon'

// L
export * from '@chakra-ui/vue-layout'

// M
export * from '@chakra-ui/c-modal'
export * from '@chakra-ui/c-motion'

// P
export * from '@chakra-ui/c-popper'
export * from '@chakra-ui/c-portal'

// R
export * from '@chakra-ui/c-reset'

// S
export * from '@chakra-ui/c-spinner'
export * from '@chakra-ui/c-scroll-lock'

// T
export * from '@chakra-ui/c-theme-provider'

// V
export * from '@chakra-ui/c-visually-hidden'

// OTHERS
export * from '@chakra-ui/vue-composables'
export * from '@chakra-ui/vue-a11y'

/**
 *
 * Directives exports
 * ==================
 *
 * Dear contributors,
 *
 * Please keep these exports in Alphabetical order :)
 */
