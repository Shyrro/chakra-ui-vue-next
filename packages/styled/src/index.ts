import { createStyled } from "./styled"
import { tags } from "./tags"

export * from "./cache"

// bind it to avoid mutating the original function
const styled = createStyled.bind()

tags.forEach((tagName) => {
  // @ts-ignore
  styled[tagName] = styled(tagName)
})

export default styled
