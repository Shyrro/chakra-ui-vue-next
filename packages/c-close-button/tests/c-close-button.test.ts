import { CCloseButton } from '../src'
import { render, userEvent, testA11y } from '../../test-utils/src'
import { nextTick } from '@vue/runtime-core'

const renderComponent = (props?: any) => {
  const base = {
    components: {
      CCloseButton,
    },
    template: '<CCloseButton data-testid="closebutton" />',
    ...props,
  }
  return render(base)
}

it('should have no a11y violations', async () => {
  await testA11y(renderComponent())
})

it('should render properly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should be in the DOM', () => {
  const { getByTestId } = renderComponent()
  expect(getByTestId('closebutton')).toBeInTheDocument()
})

it('should have correct aria-label attribute', () => {
  const { getByTestId } = renderComponent()
  expect(getByTestId('closebutton')).toHaveAttribute('aria-label', 'Close')
})

it('should emit click event on click', async () => {
  const handleClick = jest.fn()
  const { getByTestId } = renderComponent({
    template: '<c-icon-button data-testid="close" @click="handleClick" />',
    setup() {
      return {
        handleClick,
      }
    },
  })

  await userEvent.click(getByTestId('close'))
  await nextTick()
  expect(handleClick).toHaveBeenCalled()
})
