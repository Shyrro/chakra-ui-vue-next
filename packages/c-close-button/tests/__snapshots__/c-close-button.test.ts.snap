// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <button
    aria-label="Close"
    class="chakra-icon-button css-fdaoze"
    data-testid="closebutton"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-13otjrl"
      focusable="false"
      viewBox="0 0 24 24"
    >
      
    
      <g
        stroke="currentColor"
        strokewidth="1.5"
      >
        
      
        <path
          d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
          fill="none"
          strokelinecap="round"
        />
        
      
        <path
          d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
          fill="currentColor"
          strokelinecap="round"
        />
        
      
        <circle
          cx="12"
          cy="12"
          fill="none"
          r="11.25"
          strokemiterlimit="10"
        />
        
    
      </g>
      
  
    </svg>
  </button>
</DocumentFragment>
`;
